### PR TITLE
Build Telnet for Apple Silicon

### PR DESCRIPTION
**Pull request by BytesGuy**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Updated Formula for Telnet to build for Apple Silicon. I was unable to run `brew audit` due to errors installing gems (will try to resolve this soon).

```
adam@Adams-Mac homebrew-core % file /usr/local/bin/telnet
/usr/local/bin/telnet: Mach-O 64-bit executable arm64
```

This is my first PR to homebrew-core, so apologies if something is not quite right!

### CODE REVIEW

File: Formula/telnet.rb [hunk 1]
```diff
@@ -25,21 +25,24 @@ def install
       ENV["SDKROOT"] = MacOS.sdk_path
       ENV["MACOSX_DEPLOYMENT_TARGET"] = MacOS.version
 
-      # Force 64 bit-only build, otherwise it fails on Mojave
-      xcodebuild "SYMROOT=build", "-arch", "x86_64"
+      arch = Hardware::CPU.arm? ? "arm64" : "x86_64"
+
+      xcodebuild "SYMROOT=build", "-arch", arch
```
Comment (Bo98) [inline Formula/telnet.rb:30]: Might be cleaner to try (untested):

```suggestion
      xcodebuild "SYMROOT=build", "-arch", Hardware::CPU.arch
```

(and similarly for `RC_ARCHS`)

**Comment by Bo98** [comment]
> I was unable to run brew audit due to errors installing gems (will try to resolve this soon).

Yeah there's a couple issues there. One being a bug with system Ruby on Apple Silicon that has been reported to Apple.

**Comment by claui** [comment]
> errors installing gems (will try to resolve this soon)
Some analysis has been done in https://github.com/Homebrew/brew/issues/7857 (look for FB7836181 there). It boils down to `mkmf` calling clang with `-arch=arm64e` while `-arch=arm64` would have produced a working binary.

No idea how to coerce `mkmf` into using the proper architecture flavour.

**Comment by BytesGuy** [comment]
@Bo98 @claui Thanks, will take a look into that!

File: Formula/telnet.rb (hunk 1 above)
Comment (BytesGuy) [reply Formula/telnet.rb:30]: Good idea, will give this a test 👍

File: Formula/telnet.rb [hunk 2]
```diff
@@ -39,7 +38,7 @@ def install
                    "DSTROOT=build/Archive",
                    "CFLAGS=$(CC_Flags) -isystembuild/Products/",
                    "LDFLAGS=$(LD_Flags) -Lbuild/Products/",
-                   "RC_ARCHS=x86_64", # Force 64-bit build for Mojave
+                   "RC_ARCHS=%s" % Hardware::CPU.arch,
```
Comment (Bo98) [inline Formula/telnet.rb:41]: `brew style` prefers it like this:
```suggestion
                   "RC_ARCHS=#{Hardware::CPU.arch}",
```

Comment (BytesGuy) [reply Formula/telnet.rb:41]: Got it, thanks 👍 Will bear that in mind for next time!

**Review by Bo98 — APPROVED** [review]
Thanks!

**Pull request merged**